### PR TITLE
improve slack search tracking

### PR DESCRIPTION
**Pull request by camhashemi**
@leogrim wasn't a big refactor, thanks for pushing that. I still don't know what we should compute in terms of tracking parameters.

### CODE REVIEW

File: kifi-backend/modules/search/app/com/keepit/search/tracking/SearchAnalytics.scala [hunk 1]
```diff
@@ -137,16 +138,21 @@ class SearchAnalytics @Inject() (
     airbrake: AirbrakeNotifier) extends Logging {
 
   def searched(
-    userId: Id[User],
+    kifiOrSlackUsername: Either[Id[User], SlackUsername],
```
Comment (camhashemi) [inline kifi-backend/modules/search/app/com/keepit/search/tracking/SearchAnalytics.scala]: I went with `SlackUsername` as the `NonUserIdentifier` field, we're already tracking slack user id in a separate field. But we could switch those two, go with `SlackUserId` here and track the username separately.

Comment (camhashemi) [inline kifi-backend/modules/search/app/com/keepit/search/tracking/SearchAnalytics.scala]: going with `SlackUserId` instead so we can do the conversion on heimdal to a `UserEvent`.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 2]
```diff
@@ -87,7 +87,7 @@ class OrganizationCommanderImpl @Inject() (
     val badSiteUrl = normalizedSiteUrl.exists(URI.parse(_).isFailure)
     (badName, badSiteUrl) match {
       case (true, _) => Some(OrganizationFail.INVALID_MODIFY_NAME)
-      case (_, true) => Some(OrganizationFail.INVALID_MODIFY_PERMISSIONS)
+      case (_, true) => Some(OrganizationFail.INVALID_MODIFY_SITEURL)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:90]: :+1:

File: kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala [hunk 3]
```diff
@@ -178,6 +179,8 @@ object Shoebox extends Service {
     def getUserPermissionsByOrgId() = ServiceRoute(POST, "/internal/shoebox/database/getUserPermissionsByOrgId")
     def getIntegrationsBySlackChannel() = ServiceRoute(POST, "/internal/shoebox/database/getIntegrationsBySlackChannel")
     def getSourceAttributionForKeeps() = ServiceRoute(POST, "/internal/shoebox/database/getSourceAttributionForKeeps")
+    def getUserIdFromSlackTeamAndUserIds(teamId: SlackTeamId, slackUserId: SlackUserId) = ServiceRoute(GET, "/internal/shoebox/database/getUserIdFromSlackTeamAndUserIds", Param("teamId", teamId.value), Param("slackUserId", slackUserId.value))
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala:182]: Actually I've introduced a cached auth route that does just that: `ShoeboxServiceClient.getUserIdByIdentityId(IdentityHelpers.toIdentityId(teamId, slackUserId)`

File: kifi-backend/modules/common/app/com/keepit/shoebox/ShoeboxServiceClient.scala [hunk 4]
```diff
@@ -132,7 +134,8 @@ trait ShoeboxServiceClient extends ServiceClient {
   def getUserPermissionsByOrgId(orgIds: Set[Id[Organization]], userId: Id[User]): Future[Map[Id[Organization], Set[OrganizationPermission]]]
   def getIntegrationsBySlackChannel(teamId: SlackTeamId, channelId: SlackChannelId): Future[SlackChannelIntegrations]
   def getSourceAttributionForKeeps(keepIds: Set[Id[Keep]]): Future[Map[Id[Keep], SourceAttribution]]
-
+  def getUserIdFromSlackTeamAndUserIds(teamId: SlackTeamId, slackUserId: SlackUserId): Future[Option[Id[User]]]
+  def getSlackTeamInfo(slackTeamId: SlackTeamId): Future[Option[(Id[Organization], SlackTeamName)]]
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/shoebox/ShoeboxServiceClient.scala:138]: Perhaps return something more extensible than a tuple?

File: kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala (hunk 3 above)
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/common/routes/Router.scala:182]: nice, will use that.

File: kifi-backend/modules/common/app/com/keepit/shoebox/ShoeboxServiceClient.scala (hunk 4 above)
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/shoebox/ShoeboxServiceClient.scala:138]: good idea, will do

**Comment by leogrim** [comment]
lgtm, just make sure they augmentation endpoints are cached (if you switched the first one to mine, that one is) so that we don't die from our own analytics :)

**Pull request merged**